### PR TITLE
Fix fast jit issues

### PR DESCRIPTION
**Pull request by wenyongh**

### CODE REVIEW

File: core/iwasm/fast-jit/fe/jit_emit_numberic.c [hunk 1]
```diff
@@ -885,9 +885,15 @@ compile_int_shl(JitCompContext *cc, JitReg left, JitReg right, bool is_i32)
 
     res = is_i32 ? jit_cc_new_reg_I32(cc) : jit_cc_new_reg_I64(cc);
 #if defined(BUILD_TARGET_X86_64) || defined(BUILD_TARGET_AMD_64)
-    GEN_INSN(MOV, is_i32 ? ecx_hreg : rcx_hreg, right);
-    GEN_INSN(SHL, res, left, is_i32 ? ecx_hreg : rcx_hreg);
-    GEN_INSN(MOV, ecx_hreg, ecx_hreg);
+    if (jit_reg_is_const(right)) {
```
Comment (lum1n0us) [inline core/iwasm/fast-jit/fe/jit_emit_numberic.c:888]: should move `if` branch to `DEF_UNI_INT_CONST_OPS(shl)`

File: core/iwasm/fast-jit/fe/jit_emit_numberic.c [hunk 2]
```diff
@@ -915,9 +921,15 @@ compile_int_shrs(JitCompContext *cc, JitReg left, JitReg right, bool is_i32)
 
     res = is_i32 ? jit_cc_new_reg_I32(cc) : jit_cc_new_reg_I64(cc);
 #if defined(BUILD_TARGET_X86_64) || defined(BUILD_TARGET_AMD_64)
-    GEN_INSN(MOV, is_i32 ? ecx_hreg : rcx_hreg, right);
-    GEN_INSN(SHRS, res, left, is_i32 ? ecx_hreg : rcx_hreg);
-    GEN_INSN(MOV, ecx_hreg, ecx_hreg);
+    if (jit_reg_is_const(right)) {
```
Comment (lum1n0us) [inline core/iwasm/fast-jit/fe/jit_emit_numberic.c:924]: should move `if` branch to `DEF_UNI_INT_CONST_OPS(shrs)`

File: core/iwasm/fast-jit/fe/jit_emit_numberic.c [hunk 3]
```diff
@@ -945,9 +957,15 @@ compile_int_shru(JitCompContext *cc, JitReg left, JitReg right, bool is_i32)
 
     res = is_i32 ? jit_cc_new_reg_I32(cc) : jit_cc_new_reg_I64(cc);
 #if defined(BUILD_TARGET_X86_64) || defined(BUILD_TARGET_AMD_64)
-    GEN_INSN(MOV, is_i32 ? ecx_hreg : rcx_hreg, right);
-    GEN_INSN(SHRU, res, left, is_i32 ? ecx_hreg : rcx_hreg);
-    GEN_INSN(MOV, ecx_hreg, ecx_hreg);
+    if (jit_reg_is_const(right)) {
```
Comment (lum1n0us) [inline core/iwasm/fast-jit/fe/jit_emit_numberic.c:960]: should move `if` branch to `DEF_UNI_INT_CONST_OPS(shru)`

**Comment by lum1n0us** [comment]
Shall we use `ROL`, `ROR`, `RCL` and `RCR` to optimize the current ROTR and ROTL opcode?

File: core/iwasm/fast-jit/fe/jit_emit_numberic.c (hunk 1 above)
Comment (wenyongh) [reply core/iwasm/fast-jit/fe/jit_emit_numberic.c]: Done

File: core/iwasm/fast-jit/fe/jit_emit_numberic.c (hunk 3 above)
Comment (wenyongh) [reply core/iwasm/fast-jit/fe/jit_emit_numberic.c]: Done

File: core/iwasm/fast-jit/fe/jit_emit_numberic.c [hunk 4]
```diff
@@ -1000,34 +1030,14 @@ compile_int_rotl(JitCompContext *cc, JitReg left, JitReg right, bool is_i32)
...
+    res = is_i32 ? jit_cc_new_reg_I32(cc) : jit_cc_new_reg_I64(cc);
 #if defined(BUILD_TARGET_X86_64) || defined(BUILD_TARGET_AMD_64)
     GEN_INSN(MOV, is_i32 ? ecx_hreg : rcx_hreg, right);
-    GEN_INSN(SHL, shl_res, left, is_i32 ? ecx_hreg : rcx_hreg);
-
-    GEN_INSN(MOV, is_i32 ? ecx_hreg : rcx_hreg, tmp);
-    GEN_INSN(SHRU, shr_res, left, is_i32 ? ecx_hreg : rcx_hreg);
-
-    GEN_INSN(OR, res, shl_res, shr_res);
+    GEN_INSN(MOV, res, left);
```
Comment (lum1n0us) [inline core/iwasm/fast-jit/fe/jit_emit_numberic.c:1036]: it should work well, but why `mov` firstly?

Comment (wenyongh) [reply core/iwasm/fast-jit/fe/jit_emit_numberic.c:1036]: Yes, there are bugs in codegen shift_r_r_to_r_i32/shift_r_r_to_r_i64, see the modify in codegen, we should not modify the source register (shl reg_no1_src cl will change reg_no1_src). I didn't find it before, so modify here. We can restore it after codegen issue is fixed.

Comment (wenyongh) [reply core/iwasm/fast-jit/fe/jit_emit_numberic.c]: Restored, done.

**Pull request merged**